apps/web/src/pages/ApplicationFormPage/ApplicationFormPage.tsx

```diff
@@ -0,0 +1,427 @@
+import { closestCenter, DndContext, DragEndEvent } from '@dnd-kit/core';
+import {
+  arrayMove,
+  SortableContext,
+  verticalListSortingStrategy,
+} from '@dnd-kit/sortable';
+import {
+  EducationOrEmploymentStatus,
+  Field,
+  FieldInCroatian,
+  FoundOutAboutInternshipBy,
+} from '@internship-app/types';
+import { Checkbox, FormControl, RadioGroup } from '@mui/joy';
+import Radio from '@mui/joy/Radio';
+import { Button, Input } from '@mui/material';
+import clsx from 'clsx';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { usePostIntern } from '../../api/usePostIntern';
+import { SortableField } from '../../components/SortableField';
+import classes from './index.module.css';
+
+export type FormValues = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  phoneNumber: number;
+  dateOfBirth: string;
+  fields: Field[];
+  educationOrEmploymentStatus: EducationOrEmploymentStatus;
+  highSchoolOrCollegeName: string;
+  foundOutAboutInternshipBy: FoundOutAboutInternshipBy;
+  reasonForApplying: string;
+};
+
+export const ApplicationFormPage = () => {
+  const [internFields, setInternFields] = useState<Field[]>([]);
```

Comment:
Nije idealno što logika za DND sort nije izdvojena u komponentu za sebe. Pokušao bih je izdvojiti u komponentu (npr. prima opcije, label, control) koja će također omogućiti da se ponaša kao input za react hook form.

Reply:
✅